### PR TITLE
Add deprecation logger warning to Edelman response function for deprecation in pastas 2.0

### PR DESCRIPTION
**Pull request by martinvonk**
# Short Description
Add deprecation warning for Edelman response function #475 
Edelman added to the pastas-plugins library in pull request [#9](https://github.com/pastas/pastas-plugins/pull/9)

### CODE REVIEW

**Comment by martinvonk** [comment]
Fixed the failing ruff tests with #799

File: pastas/rfunc.py [hunk 1]
```diff
@@ -1513,6 +1514,13 @@ def __init__(
         cutoff: float = 0.999,
         **kwargs,
     ) -> None:
+        warn(
+            "This response function `Edelman` will be deprecated in Pastas "
```
Comment (raoulcollenteur) [inline pastas/rfunc.py:1518]: The response function

File: pastas/rfunc.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 """This module contains all the response functions available in Pastas."""
 
 from logging import getLogger
+from warnings import warn
```
Comment (raoulcollenteur) [inline pastas/rfunc.py:5]: Please follow: https://pastas.readthedocs.io/latest/developers/code_style.html

**Review by raoulcollenteur — CHANGES_REQUESTED** [review]
Let's stick to how all other warnings are throw for consistency. Can you change to logger.warning?

**Comment by martinvonk** [comment]
I can change to `logger.warning` but we now have three methods for deprecation in the code: 

[logger.warning](https://github.com/pastas/pastas/blob/b214d48dde75d2864af6d8bf51bb158103be4c52/pastas/version.py#L50-L59)
[PastasDeprecationWarning decorator](https://github.com/pastas/pastas/blob/b214d48dde75d2864af6d8bf51bb158103be4c52/pastas/decorators.py#L65-L73)
[FutureWarning](https://github.com/pastas/pastas/blob/b214d48dde75d2864af6d8bf51bb158103be4c52/pastas/stats/metrics.py#L690C1-L694C6)

Maybe we could change the PastasDeprecationWarning to call the `__name__` of the function/class and have an argument for the version when something will be deprecated.

**Review by dbrakenhoff — APPROVED** [review]

**Pull request merged**